Automattic/wp-calypso
# Analytics: remove Redux indirection when getting superProps for a tracks event

When sending a tracks event, the analytics library calculates some additional props for the tracks event that are always the same. And for `calypso_page_view` events, adds some info about the selected site to the event if the recorded path has a site parameter (like `/posts/:site`). That feature is called "super props" and the additional props are provided by the `getAll` callback in `analytics/lib/super-props`.

This PR removes several levels of Redux indirection where `recordEvent` would dispatch a `ANALYTICS_SUPER_PROPS_UPDATE` action, a middleware would intercept that action, retrieve site info from Redux and call `analytics.setSelectedSite` and `analytics.setSiteCount`. Then, after the `dispatch` call, the analytics module has a new `_selectedSite` and `_siteCount` internal variable values.

After this patch, the `superProps` getter has the `reduxStore` available and can be called directly.

**Drive-by fixes:**
- simplify the library initialization: `analytics.initialize( ... )` works both in logged-out and logged-in mode, and `analytics.setDispatch` is not needed.
- during the initialization, the analytics library tries to detect if our stats scripts are blocked by a browser extension like uBlock and bumps a stat. (Done by @gravityrail in #17688) Before this patch, there was race condition where `checkForBlockedTracks` could be called before the
`loadScript` call is finished, and `_loadTracksError` wasn't set yet. Now we wait for the `loadScript` result. See 897fcfb, could be a standalone PR.

This PR is a very nice steps towards removing unneeded dependencies from analytics and making it a self-contained and reusable library.

**How to test:**
Enable tracks debugging in console:
```js
localStorage.debug = 'calypso:analytics:tracks'
```
and watch tracks events logged as you navigate Calypso. Check that `calypso_page_view` in site-specific section includes a `blog_id` prop, and it doesn't for other events. See also #23815.
